### DISCUSSION TITLE
How to determine roborock s5 firmware version, before rooting

### DISCUSSION THREAD

**Discussion by dantheman3333** [Hypfer/Valetudo#1955] [Q&A / Support] [answered]
Hi, I've factory reset my roborock s5, and I am on this choice:
```
Select the version of the generated firmware:
S5 (ver 1886, 07/2019, full Ubuntu) update to this version first, if your current is < 1886
S5 (ver 2008, 12/2019, stripped-Ubuntu) update to this version first, if your current is < 2008
S5 (ver 2034, 11/2020, stripped-Ubuntu) latest (do not update directly to this version, if your current version is too old)


Click here to confirm that you own the device model that uses this firmware.
```
I have two choices for a starting firmware version to use, but I do not know what my current is. It has been years since I've used the app (2020) and I do not want to install it again. I have scanned the docs but could not find a way to determine the current version is. Before rooting, is there a password-based ssh auth I can use to get on the device and see the firmware version? Do I need to go through the disassembly and uart route? Or are there differences in what the robot says when pressing the buttons across versions?

**Comment by stefanmd023** [comment]
What is the manufacturing date on the robot?

**Reply by dantheman3333** [reply]
thanks for your reply, but I just went with the brute force method Hypfer suggested

**Comment by Hypfer** [comment, marked as answer]
When in doubt, just start with 1886.
Also note that only the first install after the factory reset can be done via the OTA tool. Any subsequent firmware update will be done via ssh.

**Reply by dantheman3333** [reply]
1886 didn't work, then tried 2008 and it worked, thanks. The `Click here to confirm that you own the device model that uses this firmware.` made me worried I would brick it by using the the wrong one. Perhaps the wording can be tweaked for future users
